torch/distributed/rpc/rref_proxy.py

```diff
@@ -15,24 +15,66 @@ def _local_invoke_async_execution(rref, func_name, args, kwargs):
 def _invoke_rpc(rref, rpc_api, func_name, timeout, *args, **kwargs):
     # Since rref._get_type can potentially issue an RPC, it should respect the
     # passed in timeout here.
-    rref_type = rref._get_type(timeout=timeout)
-
-    _invoke_func = _local_invoke
-    # Bypass ScriptModules when checking for async function attribute.
-    bypass_type = issubclass(rref_type, torch.jit.ScriptModule) or issubclass(
-        rref_type, torch._C.ScriptModule
-    )
-    if not bypass_type:
-        func = getattr(rref_type, func_name)
-        if hasattr(func, "_wrapped_async_rpc_function"):
-            _invoke_func = _local_invoke_async_execution
-
-    return rpc_api(
-        rref.owner(),
-        _invoke_func,
-        args=(rref, func_name, args, kwargs),
-        timeout=timeout
-    )
+    # if rpc_sync, allow get_type to block, else, add invoke logic as a callback
+    block_on_type = rpc_api in [torch.distributed.rpc.rpc_sync, torch.distributed.rpc.remote]
```

Comment:
Why does `remote` block on type as well? Shouldn't `remote` be async as well?

Reply:
I think we have to block for remote since if we run this as a callback, we would need to return a "future" to an RRef but this API expects an RRef to be returned. 

If we have a future f and we need f.result() to create an RRef, I'm not sure we currently have a good way other than:
```
f.wait()
return RRef(f.result())
```

which is what we do here, although it's blocking